### ISSUE TITLE
Creative-control activates when opening inventory for the first time while in creative after receiving a key.

### ISSUE THREAD

**Issue by mibby** [Hazebyte/CrateReloaded#271]
CrateReloaded 2.0 ALPHA3 (v3 posted late 07/20)

```
# Creative Control
# Prevents players from duplicating keys with creative gamemode.
creative-control: true
```

When you receive a key while in creative (/cr give to <name> <crate> 1), opening your inventory outputs ```Error: You may not perform thbis action in creative.``` However attempting to open your inventory a second time, this error does not occur. It is only on the first inventory open attempt after receiving a key.

**Comment by imWillX** [comment]
I'm not able to replicate this. The message is only outputted on an InventoryCreativeEvent.

**Comment by mibby** [comment]
Perhaps it is an issue with Paper? (Spigot fork) Though it wasn't an issue with CrateReloaded 1.x.

I had the error occur after I logged in and opened my inventory for the first time, while having keys in my hotbar (in creative gamemode).

**Comment by imWillX** [comment]
I don't think it's an issue with Paper. My test server is running Paper b1509.

**Comment by mibby** [comment]
What changed with creative-control between v1 and v2 internally?

**Comment by imWillX** [comment]
InventoryClickEvent (v1) vs InventoryCreativeEvent (v2)

**Comment by mibby** [comment]
Hmm. I'm not sure why you aren't able to reproduce, it happens quite frequently and repeatedly for me. Is there any downside to switching back to InventoryClickEvent?

**Comment by imWillX** [comment]
No, let's try it. `alpha4` dev build released.

**Comment by imWillX** [comment]
@mibby Fixed?

**Comment by mibby** [comment]
@imWillX Unfortunately it does not seem fixed with 2.0-ALPHA4v1 (ALPHA5 in plugin.yml). 

I gave myself a key via console ```/cr give to mibby VoterCrate 1```, then opened my inventory (while in creative). Immediately got the ```Error: You may not perform this action in creative.```. 

Did you replace all the CreativeControl events back to what existed in 1.x?

**Comment by mibby** [comment]
@imWillX Doing some further testing, this seems to be an issue with the MC 1.13 client. I can not reproduce the error occurring when _just_ opening my inventory while on MC 1.12.2.
